model/src/main/java/io/spine/validation/DefaultValue.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.validation;
+
+import com.google.protobuf.ByteString;
+import io.spine.protodata.Field;
+import io.spine.protodata.PrimitiveType;
+import io.spine.protodata.Type;
+import io.spine.protodata.Type.KindCase;
+import io.spine.protodata.TypeName;
+import io.spine.util.Exceptions;
+import org.jetbrains.annotations.NotNull;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.protodata.PrimitiveType.PT_UNKNOWN;
+import static io.spine.protodata.PrimitiveType.TYPE_BYTES;
+import static io.spine.protodata.PrimitiveType.TYPE_STRING;
+import static io.spine.protodata.PrimitiveType.UNRECOGNIZED;
+import static io.spine.util.Exceptions.newIllegalArgumentException;
+import static io.spine.util.Exceptions.newIllegalStateException;
+
+/**
+ * A factory of {@link Value}s representing default states of Protobuf message fields.
```

Comment:
As discussed vocally, let's expand this documentation piece.

In particular, we need to describe the difference between this factory and `io.spine.protobuf.Messages` (which on the surface looks more promising, equipped with some caching strategy).